lib/driver.js

```diff
@@ -352,26 +354,52 @@ class YouiEngineDriver extends BaseDriver {
         {logger.info('Connecting to WebDriver: ' + HOST + ':' + PORT);}
 
         this.socket = new net.Socket();
+        this.socket.setTimeout(5000);
+        this.socket.setKeepAlive(true, 1000);
+
+        let socketClient = this.socket;
+
+        let removeListenerHandler = function () {
+          socketClient.removeListener('timeout', timeoutHandler);
+          socketClient.removeListener('close', closeHandler);
+          socketClient.removeListener('end', endHandler);
+          socketClient.removeListener('error', errorHandler);
+        };
 
         // Add an 'error' event handler for the client socket
-        this.socket.on ('error', function (ex) {
+        let errorHandler = function (ex) {
           logger.error(ex);
           logger.error('Check that WebDriver is enabled in application, if a device ensure the proper IP address is used.');
+          removeListenerHandler();
+          socketClient.destroy(); //TODO: should we use or add socketClient.end();
           resolve(false);
-        });
+        };
+        this.socket.on ('error', errorHandler);
         // Add a 'close' event handler for the client socket
-        this.socket.on ('close', function () {
+        let closeHandler = function () {
           logger.info('Connection closed');
-        });
+          removeListenerHandler();
```

Comment:
should we use or add socketClient.end() in the close Handler?

Reply:
not needed, `close()` is called only all connections have been ended

Reply:
not needed, `close()` is called only after all connections have ended